appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/writer/PrometheusExporter.java

```diff
@@ -323,25 +323,13 @@ private double getConversionFactor(String unit) {
         } else {
             switch (unit) {
                 case NANOSECONDS:
-                    conversionFactor = NANOSECOND_CONVERSION;
-                    break;
                 case MICROSECONDS:
-                    conversionFactor = MICROSECOND_CONVERSION;
-                    break;
                 case MILLISECONDS:
-                    conversionFactor = MILLISECOND_CONVERSION;
-                    break;
                 case SECONDS:
-                    conversionFactor = SECOND_CONVERSION;
-                    break;
                 case MINUTES:
-                    conversionFactor = MINUTE_CONVERSION;
-                    break;
                 case HOURS:
-                    conversionFactor = HOUR_CONVERSION;
-                    break;
                 case DAYS:
-                    conversionFactor = DAY_CONVERSION;
+                    conversionFactor = NANOSECOND_CONVERSION;
```

Comment:
`Second` is the base time unit in the Prometheus format, instead of returning `NANOSECOND_CONVERSION` for all, May you fix the conversion factor like `NANOSECOND_CONVERSION = 1d`, `MICROSECOND_CONVERSION = 1d / 1_000d`, `MILLISECOND_CONVERSION = 1d / 1_000_000d`, ... etc in `PrometheusExporter`.

Reply:
The conversion here is from nanoseconds to seconds, both of which are currently fixed. The other time conversion constants would be correct if the time would be stored internally in the units given (e.g. the unit parameter to @Metric).

As it stands, `fish.payara.microprofile.metrics.impl.TimerImpl` always stores its final measured values in the histogram as nanoseconds, and converts the value given to the public method `update(long, TimeUnit)` to nanoseconds as well:
```
@Override
public void update(long duration, TimeUnit unit) {
    update(unit.toNanos(duration));
}
```

Reply:
Makes sense but histogram conversion factor should not be changed. I have updated the PR, please review it!

Reply:
I have verified that it works correctly for both timer and histogram, with Prometheus and JSON output.